### PR TITLE
docs:  delete Mac-related files

### LINKED ISSUE #10893: docs: Mac related documents need to be modified / deleted
### Current State

https://github.com/apache/apisix/pull/10864
We have removed some support for building and deploying APISIX on Mac in this pr, but there are still some documents that need to be update/delete.

1. https://github.com/Vacant2333/apisix/blob/ed90232a15258a9a5d225ca7f7b00e2788a3edf7/docs/en/latest/external-plugin.md?plain=1#L72
image

2. https://github.com/Vacant2333/apisix/blob/ed90232a15258a9a5d225ca7f7b00e2788a3edf7/docs/en/latest/building-apisix.md?plain=1#L82
image

### Desired State

Remove the content of support Build/Install APISIX on MacOS

### PR DESCRIPTION
**Pull request by opencmit2**
### Description
Modify/delete Mac-related files

Fixes #10893 

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: docs/en/latest/building-apisix.md [hunk 1]
```diff
@@ -69,43 +69,6 @@ make install
 
 This will install the runtime-dependent Lua libraries and `apisix-runtime` the `apisix` CLI tool.
 
-:::note
-
-If you get an error message like `Could not find header file for LDAP/PCRE/openssl` while running `make deps`, use this solution.
-
-`luarocks` supports custom compile-time dependencies (See: [Config file format](https://github.com/luarocks/luarocks/wiki/Config-file-format)). You can use a third-party tool to install the missing packages and add its installation directory to the `luarocks`' variables table. This method works on macOS, Ubuntu, CentOS, and other similar operating systems.
```
Comment (monkeyDluffy6017) [inline docs/en/latest/building-apisix.md:76]: I think we should keep this?

File: docs/zh/latest/building-apisix.md [hunk 2]
```diff
@@ -141,15 +103,6 @@ tar -xvf etcd-v${ETCD_VERSION}-linux-amd64.tar.gz && \
...
 
-</TabItem>
-
-<TabItem value="mac">
-
-```shell
-brew install etcd
-brew services start etcd
-```
-
```
Comment (Vacant2333) [inline docs/zh/latest/building-apisix.md:152]: we may need save this one, user can build and run apsix at docker, he will need install etcd by brew

File: docs/en/latest/building-apisix.md [hunk 3]
```diff
@@ -69,43 +69,6 @@ make install
...
-   2. You can also change the default configuration file of `luarocks`. Open the file `~/.luaorcks/config-5.1.lua` and add the following:
-
-      ```shell
-      variables = { LDAP_DIR = "/opt/homebrew/cellar/openldap/2.6.1", LDAP_INCDIR = "/opt/homebrew/cellar/openldap/2.6.1/include", }
-      ```
-
-      `/opt/homebrew/cellar/openldap/` is default path `openldap` is installed on Apple Silicon macOS machines. For Intel machines, the default path is  `/usr/local/opt/openldap/`.
-
-:::
-
```
Comment (Vacant2333) [inline docs/en/latest/building-apisix.md:108]: this contents should'nt be removed, the user may happen this problem at other platform, we can provide a solution at another platform replace macos, eg. Ubuntu20.04(but u may need reproduce it first)

**Review by Vacant2333 — CHANGES_REQUESTED** [review]
tnahks, and did u find any other content related of macos? we also need remove them

File: docs/en/latest/building-apisix.md (hunk 3 above)
Comment (opencmit2) [reply docs/en/latest/building-apisix.md:108]: @Vacant2333 
I have checked the history of this issue, and it only occurred on Mac, so I don't think it's necessary to keep it.

File: docs/zh/latest/building-apisix.md (hunk 2 above)
Comment (opencmit2) [reply docs/zh/latest/building-apisix.md:152]: I think if you build with Docker, you can also run etcd with Docker. Wouldn't it be better to put Docker's documentation together?

File: docs/en/latest/building-apisix.md (hunk 3 above)
Comment (monkeyDluffy6017) [reply docs/en/latest/building-apisix.md:108]: I think this content is helpful for other systems as well, maybe you can add some notes like
```
The solution below is for macOS but it works similarly for other operating systems, and macOS is not suggested:
```

File: docs/zh/latest/building-apisix.md (hunk 2 above)
Comment (Vacant2333) [reply docs/zh/latest/building-apisix.md:152]: cc @kayx23  how do u think about this

Comment (kayx23) [reply docs/zh/latest/building-apisix.md:152]: What does this doc have to do with Docker?

Comment (kayx23) [reply docs/zh/latest/building-apisix.md:152]: So the change to me looks like, going forward:

* to build APISIX from source with everything installed on the host, only Linux is supported; details are in [this doc](https://apisix.apache.org/docs/apisix/next/building-apisix/)
* to build APISIX from source in Docker, it is OS agnostic; details are in [this doc](https://apisix.apache.org/docs/docker/build/#build-an-image-from-customizedpatched-source-code)

In that case, I agree with @monkeyDluffy6017 that the relevant etcd installation supplying building APISIX in Docker should go in the Docker doc for clarity. As you can see in the last comment, it also took me a second to figure out what the connection was between macOS etcd installation and building in Docker, as two docs are separate.

**Review by shreemaan-abhishek — COMMENTED** [review]
@opencmit2 please fix the CI

File: docs/en/latest/building-apisix.md (hunk 3 above)
Comment (opencmit2) [reply docs/en/latest/building-apisix.md]: @monkeyDluffy6017 
I have retained this section of the document and added this some notes .   

reviewed the historical issues and noticed that only Mac users have encountered this problem.  I don't fully comprehend the reference value for other system。

**Comment by monkeyDluffy6017** [comment]
please make the ci pass

File: docs/en/latest/building-apisix.md (hunk 3 above)
Comment (monkeyDluffy6017) [reply docs/en/latest/building-apisix.md]: Users may meet the problem in other systems too, we can keep this for now

**Comment by opencmit2** [comment]
@monkeyDluffy6017 @shreemaan-abhishek  Could you please approve the CI

**Comment by monkeyDluffy6017** [comment]
please fix the doc lint error in ci

**Comment by opencmit2** [comment]
@monkeyDluffy6017 @shreemaan-abhishek Could you please approve the CI
image

**Review by monkeyDluffy6017 — APPROVED** [review]

File: docs/en/latest/building-apisix.md [hunk 4]
```diff
@@ -128,7 +128,6 @@ APISIX uses [etcd](https://github.com/etcd-io/etcd) to save and synchronize conf
   defaultValue="linux"
   values={[
     {label: 'Linux', value: 'linux'},
-    {label: 'macOS', value: 'mac'},
```
Comment (Vacant2333) [inline docs/en/latest/building-apisix.md:131]: looks there is 1 tab only, we may can  remove the tab?

**Review by Vacant2333 — CHANGES_REQUESTED** [review]

**Comment by monkeyDluffy6017** [comment]
@kayx23 please review this pr

File: docs/en/latest/building-apisix.md [hunk 5]
```diff
@@ -75,7 +75,7 @@ If you get an error message like `Could not find header file for LDAP/PCRE/opens
 
 `luarocks` supports custom compile-time dependencies (See: [Config file format](https://github.com/luarocks/luarocks/wiki/Config-file-format)). You can use a third-party tool to install the missing packages and add its installation directory to the `luarocks`' variables table. This method works on macOS, Ubuntu, CentOS, and other similar operating systems.
 
-The solution below is for macOS but it works similarly for other operating systems:
+The solution below is for macOS but it works similarly for other operating systems, and macOS is not suggested:
```
Comment (kayx23) [inline docs/en/latest/building-apisix.md:78]: Have trouble understanding this. It says `the solution below is for macOS` but `macOS is not suggested (recommended)`?

Comment (opencmit2) [reply docs/en/latest/building-apisix.md:78]: @kayx23 I followed his advice.
image


I don’t know how to modify it next, so I closed it first.

**Pull request closed without merging**